### ISSUE TITLE
1.9 - Crash while launching game client, BoP implicated

### ISSUE THREAD

**Issue by mjsonofharry** [Glitchfiend/BiomesOPlenty#760]
#### Description:

I run a dedicated server, and one of my clients suffers a startup crash if he has Biomes O' Plenty installed. Concurrent mods include:

- Progressive Automation

- Just Enough Items

- Blood Magic

No other clients have reported any issues.

#### Crash Report:
```
---- Minecraft Crash Report ----
// Sorry :(

Time: 16.05.2016 05:48
Description: Initializing game

java.lang.IllegalArgumentException: Block: class biomesoplenty.common.block.BlockBOPGrass has property: variant with invalidly named value: smolderıng
	at net.minecraft.block.state.BlockStateContainer.func_185919_a(BlockStateContainer.java:113)
	at net.minecraft.block.state.BlockStateContainer.<init>(BlockStateContainer.java:73)
	at net.minecraft.block.state.BlockStateContainer.<init>(BlockStateContainer.java:58)
	at biomesoplenty.common.block.BlockBOPGrass.func_180661_e(BlockBOPGrass.java:70)
	at net.minecraft.block.Block.<init>(Block.java:205)
	at net.minecraft.block.Block.<init>(Block.java:214)
	at net.minecraft.block.BlockGrass.<init>(BlockGrass.java:25)
	at biomesoplenty.common.block.BlockBOPGrass.<init>(BlockBOPGrass.java:98)
	at biomesoplenty.common.init.ModBlocks.init(ModBlocks.java:106)
	at biomesoplenty.core.BiomesOPlenty.preInit(BiomesOPlenty.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:561)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:228)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:135)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:584)
	at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:249)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:434)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:346)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at net.minecraft.block.state.BlockStateContainer.func_185919_a(BlockStateContainer.java:113)
	at net.minecraft.block.state.BlockStateContainer.<init>(BlockStateContainer.java:73)
	at net.minecraft.block.state.BlockStateContainer.<init>(BlockStateContainer.java:58)
	at biomesoplenty.common.block.BlockBOPGrass.func_180661_e(BlockBOPGrass.java:70)
	at net.minecraft.block.Block.<init>(Block.java:205)
	at net.minecraft.block.Block.<init>(Block.java:214)
	at net.minecraft.block.BlockGrass.<init>(BlockGrass.java:25)
	at biomesoplenty.common.block.BlockBOPGrass.<init>(BlockBOPGrass.java:98)
	at biomesoplenty.common.init.ModBlocks.init(ModBlocks.java:106)
	at biomesoplenty.core.BiomesOPlenty.preInit(BiomesOPlenty.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:561)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:228)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:135)
	at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:584)
	at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:249)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:434)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:346)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.9
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 151515512 bytes (144 MB) / 386125824 bytes (368 MB) up to 1060372480 bytes (1011 MB)
	JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.23 Powered by Forge 12.16.1.1887 7 mods loaded, 7 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
	UCH	mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
	UCH	FML{8.0.99.99} [Forge Mod Loader] (forge-1.9-12.16.1.1887.jar) 
	UCH	Forge{12.16.1.1887} [Minecraft Forge] (forge-1.9-12.16.1.1887.jar) 
	UCE	BiomesOPlenty{4.0.1.2006} [Biomes O' Plenty] (BiomesOPlenty-1.9-4.0.1.2006-universal.jar) 
	UCH	JEI{3.3.3.197} [Just Enough Items] (jei_1.9-3.3.3.197.jar) 
	UCH	BloodMagic{1.9-2.0.0-38} [Blood Magic: Alchemical Wizardry] (BloodMagic-1.9-2.0.0-38.jar) 
	UCH	progressiveautomation{1.6.42} [Progressive Automation] (ProgressiveAutomation-1.9-1.6.42.jar) 
	Loaded coremods (and transformers): 
	GL info: ' Vendor: 'Intel' Version: '4.3.0 - Build 10.18.15.4248' Renderer: 'Intel(R) HD Graphics 4600'
	Launched Version: 1.9-forge1.9-12.16.1.1887
	LWJGL: 2.9.4
	OpenGL: Intel(R) HD Graphics 4600 GL version 4.3.0 - Build 10.18.15.4248, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: No
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 8x Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz
```

### Additional Information:

Something odd that I noticed in line seven:
`java.lang.IllegalArgumentException: Block: class biomesoplenty.common.block.BlockBOPGrass has property: variant with invalidly named value: smolderıng`
The word "smolderıng" contains a dotless i. I am not sure where this came from, but given that my client uses a Turkish keyboard I suspect that it is just a typo.
____
#### Affected Versions:

- Biomes O' Plenty: 1.9-4.0.1.2006
- Minecraft: 1.9
- Forge: 1.9-12.16.1.1887 and 1.9-12.16.1.1906 (crashed on both)

**Comment by lazerzes** [comment]
I believe I have found a fix, working on it now.

**Comment by lazerzes** [comment]
Fixed and Pull Request Submitted -- #768

**Comment by mjsonofharry** [comment]
I can confirm that this resolves the Turkish-I problem.
